pkg/registry/core/rest/storage_core.go

```diff
@@ -179,50 +175,90 @@ func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(restOptionsGetter generi
 
 	serviceRest := service.NewStorage(serviceRegistry, endpointRegistry, ServiceClusterIPAllocator, ServiceNodePortAllocator, c.ProxyTransport)
 
-	restStorageMap := map[string]rest.Storage{
```

Comment:
On more research and help from @liggitt What you are saying is correct: https://github.com/kubernetes/kubernetes/pull/42303#issuecomment-283230144. Will update the PR